code/game/objects/items/weapons/grenades/atmosgrenade.dm

```diff
@@ -0,0 +1,28 @@
+
+
+
+/obj/item/weapon/grenade/gas
+	name = "Plasma Fire Grenade"
+	desc = "A compressed plasma grenade, used to start horrific plasma fires."
+	origin_tech = "materials=3;magnets=4;syndicate=4"
+	icon = 'icons/obj/grenade.dmi'
+	icon_state = "syndicate"
+	item_state = "flashbang"
+
+/obj/item/weapon/grenade/gas/prime()
+	var/turf/target_turf = get_turf(src)
+	if(istype(target_turf))
+		target_turf.atmos_spawn_air("plasma=100;TEMP=1000")
+		target_turf.air_update_turf()
+	qdel(src)
+
+/obj/item/weapon/grenade/gas/knockout
+	name = "Knockout Grenade"
+	desc = "A grenade that completely removes all air and heat from its detonation area."
+
+/obj/item/weapon/grenade/gas/knockout/prime()
+	var/turf/target_turf = get_turf(src)
+	if(istype(target_turf))
+		target_turf.atmos_spawn_air("n2o=100;TEMP=[T20C]")
+		target_turf.air_update_turf()
+	qdel(src)
```

Comment:
try to find a way to have both grenade use the same prime(). This is pretty much a copypasta.